lambda/voting/frontend/src/Components/Voting/Voting.js

```diff
@@ -51,46 +54,83 @@ export const Voting = ({ sdk }) => {
     }
   };
 
-  return !result ? (
-    <div className="voting">
-      <h1>Vote for Your Favourite Candidate!</h1>
-      <form onSubmit={handleSubmit}>
-        <div className="form-group">
-          <label htmlFor="voterId">Candidate Email Id</label>
-          <input
-            type="text"
-            className="form-control"
-            id="voterId"
-            placeholder="Enter Voter Email"
-            value={voterId}
-            onChange={(e) => setVoterId(e.target.value)}
-          />
-        </div>
-        <div className="form-group">
-          <label htmlFor="asEmail">Your Email</label>
-          <input
-            type="text"
-            className="form-control"
-            id="asEmail"
-            placeholder="Enter Your Email"
-            value={asEmail}
-            onChange={(e) => setAsEmail(e.target.value)}
-          />
+  let getCandidateDetails = async (event) => {
+    event.preventDefault();
+    try {
+      setLoading(true);
+      const getCandidateDetailsResponse = await getCandidature(voterId);
+      setLoading(false);
+      if (getCandidateDetailsResponse.status !== "SUCCESS") {
+        alert(
+          "Error getting candidate details: " +
+            JSON.stringify(getCandidateDetailsResponse)
+        );
+        return;
+      }
+      setCandidateDetails(getCandidateDetailsResponse);
+    } catch (error) {
+      setLoading(false);
+      alert(
+        "Error getting Candidate Details:" +
+          (error.message ? error.message : JSON.stringify(error))
+      );
+    }
+  };
+
+  if (!candidateDetails) {
+    return (
+      <div className="voting">
+        <h1>Vote Candidature</h1>
+        <form onSubmit={getCandidateDetails}>
+          <label>
+            Candidate Email:
+            <input
+              type="text"
+              value={voterId}
+              onChange={(e) => setVoterId(e.target.value)}
+            />
+          </label>
+          <input type="submit" value="Submit" />
+        </form>
+      </div>
+    );
+  }
+
+  if (!result) {
```

Comment:
what is happening here?

Reply:
here it just opens a candidate details so the user can vote. and once transaction is queued, the result would store the transaction details and it will be shown on the page.